lib/gitlab/configuration.rb

```diff
@@ -32,6 +32,7 @@ def options
     def reset
       self.endpoint       = ENV['GITLAB_API_ENDPOINT']
       self.private_token  = ENV['GITLAB_API_PRIVATE_TOKEN']
+      self.auth_token     = ENV['GITLAB_API_AUTH_TOKEN']
```

Comment:
For users which keep using their `private_token` nothing changes. I am sorry, I don't how adding a new key for a different authentication method could cause confusion?